### PR TITLE
Add an index.d.ts for TypeScript users of the library

### PR DESCRIPTION
**Pull request by airhorns**
Thanks so much for this handy little library! I use TypeScript for most JS devs and wanted to see if you'd be open to including TypeScript type definitions for the library right in the code. If they rest here next to the code then it's a bit easier to keep them up to date and find them for anyone who needs them. If not -- no worries, I could put them in the community maintained third-party types repo over at DefinitelyTyped.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+declare module "use-bus" {
+  export interface EventAction {
+    type: string;
+    [key: string]: any;
+  }
+  export function dispatch(name: string): void;
+  export function dispatch(payload: EventAction): void;
```
Comment (fabienjuif) [inline index.d.ts]:
```suggestion
  export function dispatch(event: EventAction): void;
```
WDYT?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  export default function useBus(name: string, callback: (payload: EventAction) => void, deps: any[]): typeof dispatch;
```
Comment (fabienjuif) [inline index.d.ts]:
```suggestion
  export default function useBus(name: string, callback: (event: EventAction) => void, deps: any[]): typeof dispatch;
```

**Comment by fabienjuif** [comment]
This is a good idea, thank you for this PR.
I make some suggestions, feel free to argue!

Did you see the other library I pushed some day ago? https://github.com/fabienjuif/myrtille

File: package.json [hunk 3]
```diff
@@ -5,5 +5,9 @@
   "module": "index.js",
   "repository": "git@github.com:fabienjuif/use-bus.git",
   "author": "Fabien JUIF <fabien.juif@gmail.com>",
-  "license": "MIT"
+  "license": "MIT",
+  "types": "index.d.ts",
+  "dependencies": {
```
Comment (fabienjuif) [inline package.json]:
```suggestion
  "peerDependencies": {
```

Comment (fabienjuif) [reply package.json]: Could you re-run yarn after this one?

Comment (airhorns) [reply package.json]: Yep!

**Comment by airhorns** [comment]
No I hadn't seen that actually, looks great! `useBus` is so dead simple that it meets my needs at the moment but it's great that there's another step to take before having to go for a full state manager or redux or what have you. 💯 for small sharp tools!

**Comment by fabienjuif** [comment]
Thank you :)
Have you done there?

If yes, I'll merge and publish a new version ASAP

**Comment by airhorns** [comment]
Yep all done, thanks!

**Comment by fabienjuif** [comment]
You can use version 2.1.0 :)

**Pull request merged**